src/cldfbench/commands/media.py

```diff
@@ -0,0 +1,376 @@
+    )
+    parser.add_argument(
+        '-l', '--list',
+        help='List available mimetypes and file number and size',
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        '-o', '--out',
+        help='Directory to which to download the media files and to create the to be released data.',
+        type=PathType(type='dir'),
+        default=pathlib.Path('.')
+    )
+    parser.add_argument(
+        '-c', '--communities',
+        default='',
+        help='Comma-separated list of communities to which the dataset should be submitted',
+    )
+    parser.add_argument(
+        '-p', '--parent-doi',
+        default='',
+        help='DOI to which this release refers (format 10.5281/zenodo.1234567). It is required for --create-release.',
+    )
+    parser.add_argument(
+        '--create-release',
+        help='Switch to create ID_{0} directory containing {0}.zip, README.md and {1} for releasing on zenodo. Cannot be used with --update-zenodo.'.format(
+            MEDIA, ZENODO_FILE_NAME),
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        '--update-zenodo',
+        help="Deposit ID (number after DOI's last slash) to update metadata by using ID_{0}/{1}. Cannot be used with --create-release.".format(
+            MEDIA, ZENODO_FILE_NAME),
+        default=None,
+    )
+    parser.add_argument(
+        '--debug',
+        help='Switch to work with max. 500 media files and with sandbox.zenodo for testing ONLY',
+        action='store_true',
+        default=False,
+    )
+
+
+def _create_download_thread(url, target):
+    global download_threads
+    download_threads = []
+
+    def _download(url, target):
+        assert not target.exists()
+        urlretrieve(url, str(target))
+
+    while threading.active_count() > 7:
+        time.sleep(0.1)
+
+    download_thread = threading.Thread(target=_download, args=(url, target))
+    download_thread.start()
+    download_threads.append(download_thread)
+
+
+def _git_repo_url(dir_, git_command='git'):
+    dir_ = pathlib.Path(dir_)
+    if not dir_.exists():
+        raise ValueError('cannot get repo data of non-existent directory')
+    dir_ = dir_.resolve()
+    cmd = [git_command, '--git-dir=%s' % dir_.joinpath('.git'), 'config', '--get', 'remote.origin.url']
+    try:
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        if p.returncode == 0:
+            res = stdout.strip()  # pragma: no cover
+        else:
+            raise ValueError(stderr)
+    except (ValueError, FileNotFoundError):
+        res = dir_.name
+    if not isinstance(res, str):
+        res = res.decode('utf8')
+    return res.replace('.git', '')
+
+
+def run(args):
+
+    ds = get_dataset(args)
+    ds_cldf = ds.cldf_reader()
+    release_dir = args.out / '{0}_{1}'.format(ds.id, MEDIA)
+
+    if ds_cldf.get('media.csv', None) is None:
```

Comment:
I think for all these preliminary input checks, I'd raise [`ParserError`](https://github.com/clld/clldutils/blob/67a8c493c2bbf37c107ebae72bbe80e8d6a2a1a3/src/clldutils/clilib.py#L21).